ca_tools/scripts/pid_controller.py

```diff
@@ -0,0 +1,238 @@
+
+        #suscribers and publishers
+        self._my_pid_angular_state_pub = rospy.Publisher(
+            "/angular/state", Float64, queue_size=1)
+        self._my_pub = rospy.Publisher(pubtopic, Twist, queue_size=10)
+        self._my_pid_lineal_state_pub = rospy.Publisher(
+            "/lineal/state", Float64, queue_size=1)
+        self._my_pid_angular_effort_sub = rospy.Subscriber(
+            "/lineal/control_effort", Float64, self._lineal_effort_callback)
+        self._my_pid_lineal_setpoint_pub=rospy.Publisher("/lineal/setpoint",Float64,queue_size=10)
+        self._my_sub = rospy.Subscriber(subtopic, Odometry, self._callback)
+        self._my_pid_angular_effort_sub = rospy.Subscriber(
+            "/angular/control_effort", Float64, self._angular_effort_callback)
+        self._my_path_publisher = rospy.Publisher(
+            "/create1/my_path", Path, queue_size=10)
+        self._my_path = Path()
+        self._my_pid_angular_setpoint_pub = rospy.Publisher("/angular/setpoint", Float64, queue_size=1)
+
+        self._my_goals = []
+        self._path_threshold = 60 #Throttle the path pub
+        #CREATE GOALS
+        for i in range(4):
+            self._my_goals.append(Pose2D())
+        self._my_goals[1].x = 2.0
+        self._my_goals[2].x = 2.0
+        self._my_goals[2].y = 2.0
+        self._my_goals[3].y = 2.0
+
+        self._lineal_state=Float64()
+        self._goal_num = 0
+        self._state = STOP_FORWARD
+        self._my_pose = Pose2D()
+        self._my_pose.x = "unset"
+        self._my_pose.y = "unset"
+        self._my_pid_state = 0.0
+        self._my_angle_goal = 0.0
+        self._my_pid_angular_effort = 0.0
+        self._my_pid_lineal_effort = 0.0
+
+    def _angular_effort_callback(self, data):
+        """Callback for the angular effort calculated by the PID node
+
+        Arguments:
+            data {[Float64]} -- [Effort]
+        """
+        self._my_pid_angular_effort = data.data
+    
+    def _lineal_effort_callback(self, data):
+        """Callback for the angular effort calculated by the PID node
+
+        Arguments:
+            data {[Float64]} -- [Effort]
+        """
+        self._my_pid_lineal_effort = data.data
+
+
+    def _callback(self, data):
+        """Callback for groundtruth
+
+        Arguments:
+            data {[Odometry]} -- [Odometry msg from gts]
+        """
+        self._my_pose.x = data.pose.pose.position.x  # Update pose
+        self._my_pose.y = data.pose.pose.position.y
+        q = data.pose.pose.orientation  # get orientation quaternion
+        self._path_threshold -= 1
+        if (self._path_threshold == 0):  # add the pose to path every 60 cycles
+            aux = PoseStamped()
+            aux.pose = data.pose.pose
+            aux.header.frame_id = "/map"
+            self._my_path.header.frame_id = "/map"
+            self._my_path.poses.append(aux)
+            self._my_path_publisher.publish(self._my_path)
+            self._path_threshold = 60
```

Comment:
Using this in more than one place, you might want to define a constant with this value.